src/main/java/com/epam/page/object/generator/models/searchrule/CommonSearchRule.java

```diff
@@ -0,0 +1,127 @@
+package com.epam.page.object.generator.models.searchrule;
+
+import com.epam.page.object.generator.errors.XpathToCssTransformerException;
+import com.epam.page.object.generator.models.ClassAndAnnotationPair;
+import com.epam.page.object.generator.models.Selector;
+import com.epam.page.object.generator.models.webgroup.CommonWebElementGroup;
+import com.epam.page.object.generator.models.webgroup.WebElementGroup;
+import com.epam.page.object.generator.models.webelement.CommonWebElement;
+import com.epam.page.object.generator.models.webelement.WebElement;
+import com.epam.page.object.generator.utils.SearchRuleType;
+import com.epam.page.object.generator.utils.SelectorUtils;
+import com.epam.page.object.generator.utils.XpathToCssTransformer;
+import com.epam.page.object.generator.validators.ValidationResult;
+import com.epam.page.object.generator.validators.ValidatorVisitor;
+import java.util.ArrayList;
+import java.util.List;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CommonSearchRule implements SearchRule {
+
+    private String uniqueness;
+    private SearchRuleType type;
+    private Selector selector;
+    private ClassAndAnnotationPair classAndAnnotation;
+    private XpathToCssTransformer transformer;
+    private SelectorUtils selectorUtils;
+
+    private List<ValidationResult> validationResults = new ArrayList<>();
+    private final static Logger logger = LoggerFactory.getLogger(CommonSearchRule.class);
+
+    public CommonSearchRule(String uniqueness, SearchRuleType type, Selector selector,
+                            ClassAndAnnotationPair classAndAnnotation,
+                            XpathToCssTransformer transformer,
+                            SelectorUtils selectorUtils) {
+        this.uniqueness = uniqueness;
+        this.type = type;
+        this.selector = selector;
+        this.classAndAnnotation = classAndAnnotation;
+        this.transformer = transformer;
+        this.selectorUtils = selectorUtils;
+    }
+
+    public String getUniqueness() {
+        return uniqueness;
+    }
+
+    public SearchRuleType getType() {
+        return type;
+    }
+
+    private String getRequiredValue(Element element) {
+        return uniqueness.equals("text")
+            ? element.text()
+            : element.attr(uniqueness);
+    }
+
+    public ClassAndAnnotationPair getClassAndAnnotation() {
+        return classAndAnnotation;
+    }
+
+    public Selector getTransformedSelector() {
+        if (!uniqueness.equalsIgnoreCase("text") && selector.isXpath()) {
+            try {
+                return transformer.getCssSelector(selector);
+            } catch (XpathToCssTransformerException e) {
+                e.printStackTrace();
```

Comment:
Use logging here instead of printStackTrace

Reply:
Done